### PR TITLE
feat(hub-common): add event entity

### PR DESCRIPTION
**Pull request by rweber-esri**
affects: @esri/hub-common

ISSUES CLOSED: 7511

1. Description:
- Adds Event entity
- Adds create Event schema & ui schema
- Adds create Event logic

1. Instructions for testing:

1. Closes Issues: #[7511](https://devtopia.esri.com/dc/hub/issues/7511) (if appropriate)

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -96,7 +96,8 @@
 		"isomorphic-form-data": "^1.0.0",
 		"node-fetch": "^2.0.1",
 		"tslib": "^1.13.0",
-		"ultra-runner": "^3.10.5"
+		"ultra-runner": "^3.10.5",
+		"dayjs": "^1.11.10"
```
Comment (rweber-esri) [inline package.json:100]: The [date-fns](https://github.com/date-fns/date-fns) package is currently a **devDependency** (only used [here](https://github.com/Esri/hub.js/blob/master/support/changelog.js#L6C29-L6C37) AFAIK), but the it doesn't support time zones out of the box; that requires also adding [date-fns-tz](https://github.com/marnusw/date-fns-tz). I tried bumping `date-fns` to latest but sadly the latest `date-fns-tz` is broken w/ the latest version of `date-fns` (also tested w/ latest `date-fns-tz: v3.0.0-beta.3`, published 4 days ago, to no avail). So adding `dayjs` as a dep to facilitate the adjustment of dates to the appropriate timezone so we post the right ISO date/time string to the server.

File: packages/common/src/events/_internal/PropertyMapper.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  IOnlineMeeting,
+} from "../api/orval/api/orval-events";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import timezone from "dayjs/plugin/timezone";
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+function getTimeZoneISOStringFromLocalDateTime(
```
Comment (rweber-esri) [inline packages/common/src/events/_internal/PropertyMapper.ts:22]: Going to move this to a util that can be used elsewhere

**Comment by rweber-esri** [comment]
Only moved to `ready for review` to get a PR-Preview I can link into my opendata-ui sandbox, still things, have a bit of cleanup to do yet and still owe tests.

**Comment by rweber-esri** [comment]
I _believe_ this is ready for review now.

File: packages/common/src/core/updateHubEntity.ts [hunk 3]
```diff
@@ -85,6 +89,18 @@ export const updateHubEntity = async (
         context.requestOptions
       );
       break;
+    case "survey":
```
Comment (juliannaeapicella) [inline packages/common/src/core/updateHubEntity.ts:92]: Thanks for catching this!

File: packages/common/src/surveys/HubSurvey.ts [hunk 4]
```diff
@@ -37,6 +38,33 @@ export class HubSurvey
     return new HubSurvey(pojo, context);
   }
 
+  /**
+   * Fetch a Survey from the backing store and return a HubSurvey instance.
+   * @param identifier - Identifier of the survey to load
+   * @param context
+   * @returns
+   */
+  static async fetch(
```
Comment (juliannaeapicella) [inline packages/common/src/surveys/HubSurvey.ts:47]: Thanks for this update! I don't remember why we didn't initially have a fetch function for surveys...

**Review by juliannaeapicella — APPROVED** [review]
LGTM! And thanks for the extra Survey entity updates!

File: packages/common/src/surveys/HubSurvey.ts (hunk 4 above)
Comment (rweber-esri) [reply packages/common/src/surveys/HubSurvey.ts:47]: Opendata-ui fetches entities using the [fetchHubEntity](https://github.com/Esri/hub.js/blob/f/7511-event-entity/packages/common/src/core/fetchHubEntity.ts) facade method which bypasses the entity classes altogether and delegates directly to the entity's fetch method in their respective `fetch` modules, so it technically didn't hurt anything for this _not_ to be implemented. I simply added it for consistency.

**Review by dbouwman — APPROVED** [review]
Thanks!

**Pull request merged**